daq_macros.py

```diff
@@ -211,9 +213,54 @@ def changeImageCenterHighMag(x,y,czoom):
   setPvDesc("highMagMinY",new_minY)    
   setPvDesc("highMagCursorX",noZoomCenterX)
   setPvDesc("highMagCursorY",noZoomCenterY)
-  
+
+sample_detection = {
+  "large_box_width": 0,
+  "large_box_height": 0,
+  "small_box_height": 0,
+  "center_x" : 0,
+  "center_y" : 0,
+  "face_on_omega" : 0
+}
+
+
+def run_top_view_optimized():
+    RE(topview_optimized())
+
+def loop_center_plan():
+    global sample_detection
+    gov_status = gov_lib.setGovRobot(gov_robot, 'SA')
+    if gov_status.success:
+      yield from detect_loop(sample_detection)
+    else:
+      print("could not trasition to SA")
 
 def autoRasterLoop(currentRequest):
+    global sample_detection, autoRasterFlag, max_col
+    RE(loop_center_plan())
```

Comment:
we should start thinking about how to better handle running plans within these functions. just thinking ahead to the future where the highest level function gets executed in the Queueserver (or its successors), and this won't be able to run because you can't have a RE(call) within a plan.

Reply:
I see this has been changed. resolving as done.